Enhancements to databases documentation

goetzk opened the pull request:
I've tried to consolidate the permissions requirements (as they're identical on
all databases) and to improve readability added some formatting to the mysql
entry.
Example commands are added for postgresql, though the section is by no means a
full guide.

goetzk committed:
Enhancements to databases documentation

I've tried to consolidate the permissions requirements (as they're identical on
all databases) and to improve readability added some formatting to the mysql
entry.
Example commands are added for postgresql, though the section is by no means a
full guide.

cbeach47 commented on guides/InstallationAdminGuide.md:
```diff
@@ -58,12 +58,13 @@ openEQUELLA installation involves the following steps:
 All databases must be able to store character data using UTF-8 encoding.
 
-A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database.
+A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database. That means the equella database user must have permission to create, modify and delete tables, indexes and constraints and to run, select, insert, delete and update queries. 
```
I would clarify `...and to run, select, insert, delete and update queries. `

to be something like the following.  as-is, makes it sound like 'run' is a query type: `...and to run the CRUD queries of: select, insert, delete and update. `

cbeach47 commented on guides/InstallationAdminGuide.md:
````diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
 
 It is required that Microsoft SQL Server databases have READ_COMMITTED_SNAPSHOT enabled to avoid possible deadlocks. See http://msdn.microsoft.com/en-us/library/ms173763.aspx for more information. The following statement will enable this setting for a given database:
 
+```
````
Add code block type:
```` ```sql ````

cbeach47 commented on guides/InstallationAdminGuide.md:
```diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
 * Database Owner: the default name used is ‘equellauser’.
 * Tablespace:‘pg_default’.
 
+These requirements are met by the following commands
```
:+1: 
Consider adding:  From a terminal (not via psql)

Also, I needed to run as postgres to make the createuser / createdb commands work.
```bash
sudo -u postgres -i
```

cbeach47 commented on guides/InstallationAdminGuide.md:
````diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
+createuser --interactive --pwprompt equellauser
+createdb -O equellauser -D pg_default -E UTF-8 equella
+```
+
````
Consider adding:  For ```createuser```, you can respond 'n' for:
- Shall the new role be a superuser? (y/n) 
- Shall the new role be allowed to create databases? (y/n) 
- Shall the new role be allowed to create more new roles? (y/n)

cbeach47 replied:
The user doesn't need to be able to create databases nor be a superuser.

cbeach47 commented on guides/InstallationAdminGuide.md:
````diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
 
+These requirements are met by the following commands
+
+```
````
Add code block type:
```` ```bash ````

goetzk replied:
I used 'sh', hopefully highlights the same but happy to change to 'bash' should it be required.

cbeach47 reviewed (changes requested):
Thanks for the consolidation and details!

I suggested some minor changes to include before we have this go live.

goetzk committed:
Change requirements section following feedback

This change attempts to clarify this section, resolving the feedback but ina
slightly different way than was suggested.
Creating a list seems like a nicer way to show the two groupings of permissions
required.

goetzk committed:
Add formatting sections to code blocks

SQL and shell entries have had their fenced code blocks changed to (hopefully)
syntax highlight.

goetzk committed:
Adjust user creation command

Rather than ask the admin to hit 'y' three times, lets just tell createuser that
we want a superuser who can create DBs.
Keeps our requried documentation smaller.

goetzk commented:
Hi @cbeach47 ,
I've made changes based on your feedback. I haven't directly copied your suggestions but the forced reviewing meant I saw some other ways to phrase things.

cbeach47 commented on guides/InstallationAdminGuide.md:
```diff
@@ -58,12 +58,16 @@ openEQUELLA installation involves the following steps:
+* Create, modify and delete tables, indexes and constraints
+* Perform select, insert, delete and update queries on the newly created databases.
+
+You must also ensure the database user has a password as openEQUELLA will request id during installation.
```
minor change -
> You must also ensure the database user has a password as openEQUELLA will request **id** during installation.

to
> You must also ensure the database user has a password as openEQUELLA will request **it** during installation.

goetzk replied:
i have this change staged locally

cbeach47 commented on guides/InstallationAdminGuide.md:
```diff
@@ -58,12 +58,16 @@ openEQUELLA installation involves the following steps:
 All databases must be able to store character data using UTF-8 encoding.
 
-A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database.
+A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database. That means the equella database user must have permission to:
```
> That means the **equella** database user must have permission to:

to
> That means the **openequella** database user must have permission to:

goetzk replied:
I have this change staged locally

cbeach47 reviewed (changes requested):
minor edits requested.

goetzk committed:
Changes per bug feedback.

goetzk committed:
Resolve issues with createuser invocation

As requested in feedback reverted change in
694dc7ab8fda4579f4635a59b327c03a66a6a9f5 to remove --superuser. Removed
--interactive; since all three prompts are 'no' we needn't ask for
input.

goetzk committed:
Merge remote-tracking branch 'upstream/master' into databases

goetzk commented:
@cbeach47 Hopefully now in a suitable state.

cbeach47 commented:
:+1:  looks good - thanks @goetzk !
